clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1285,6 +1285,8 @@ void Sema::ConstructOpenCLKernel(FunctionDecl *KernelCallerFunc,
   FunctionDecl *OpenCLKernel =
       CreateOpenCLKernelDeclaration(getASTContext(), Name, ParamDescs);
 
+  ContextRAII FuncContext(*this, OpenCLKernel);
```

Comment:
Do we really need this? We disabled diagnosing during OpenCL kernel generation, is it possible that we once enable it?

Reply:
At the moment, I no longer have a true motivating example. However, this is a bug we have worked around a few times, so I suspect we will need this fixed eventually.

Additionally, this is one of the conditions where getLexicalContext gave us not a Function Decl, so now that this is fixed, we know !FD means exactly what it should, that we are evaluating a global (and this can properly handle that once we figure that out).

Reply:
Okay, sounds reasonable for me.